chia/full_node/full_node.py

```diff
@@ -1526,6 +1527,8 @@ async def respond_block(
                     ):
                         self.full_node_store.previous_generator = None
 
+                if raise_if_not_added and added != ReceiveBlockResult.NEW_PEAK:
+                    raise RuntimeError(f"Expected block to be added, received {added}")
```

Comment:
This does override the later check for invalid_block which raises ConsensusError

Reply:
would it be better to raise it in every if brach, before the `return None`? I went for this just to introduce minimal change.

Reply:
I don't know all the nuances, but this code already has a failsafe raise at the end, so I think it makes sense to only add this to the DISCONNECTED_BLOCK section.

ALREADY_HAVE_BLOCK: (should this raise?)
INVALID_BLOCK: already raise ConsensusError
DISCONNECTED_BLOCK: `add raise`
NEW_PEAK: no raise
ADDED_AS_ORPHAN: no raise
Anything else: already raise RuntimeError

Reply:
I don't think we should raise there, it's possible a peer gives us a block that is disconnected, we don't want to disconnect. 
We should just return as I mentioned in the other comment

Reply:
fixed, I didn't initially realised it's just 1 case we're interested in.